pkg/cap/pubsub/client.go

```diff
@@ -51,8 +52,30 @@ func (t Topic) Publish(ctx context.Context, b []byte) error {
 	return err
 }
 
-func (t Topic) Subscribe(ctx context.Context) (*Subscription, error) {
-	return newSubscription(ctx, api.Topic(t), make(chan []byte, 32))
+func (t Topic) Subscribe(ctx context.Context, ch chan<- []byte) (cancel func(), err error) {
```

Comment:
`cancel func()` I would call it `release func()` to stick to capnproto terminology. I would only use cancel if we wrap the release function into a function that also calls other functions, e.g. a context cancellation.

Reply:
Yeah, I was torn between the `release` nomenclature from capnp and the `cancel` terminology libp2p's `pubsub.Subscription.Cancel`.  But from the user's perspective, "cancel a subscription" is more familiar than "release a subscription".  Also, `release` refers to low-level concerns about refcounting, which the user shouldn't need to worry about.